src/components/UserInfo.vue

```diff
@@ -156,82 +164,84 @@ export default {
+        this.user.id,
+        this.selectedClientId
+      ).catch(e => {
+        console.log(e);
+      });
     },
 
     getUserAvailableClientRoles: function() {
-      UsersRepository.getUserAvailableClientRoles(
+      return UsersRepository.getUserAvailableClientRoles(
         this.user.id,
         this.selectedClientId
-      )
-        .then(response => {
-          this.availableClientRoles.push(...response.data);
-        })
-        .catch(e => {
-          console.log(e);
-        });
+      ).catch(e => {
+        console.log(e);
+      });
     },
 
     getUserEffectiveClientRoles: function() {
-      UsersRepository.getUserEffectiveClientRoles(
+      return UsersRepository.getUserEffectiveClientRoles(
         this.user.id,
         this.selectedClientId
-      )
-        .then(response => {
-          this.effectiveClientRoles.push(...response.data);
-          this.selectedRoles.push(...response.data);
-        })
-        .catch(e => {
-          console.log(e);
-        });
+      ).catch(e => {
+        console.log(e);
+      });
     },
 
     updateUserClientRoles: function() {
-      //If in effective but not selected DELETE
-      var rolesToDelete = this.effectiveClientRoles.filter(
+      //If in roles but not selected DELETE
+      var rolesToDelete = this.clientRoles.filter(
         value => !this.selectedRoles.includes(value)
       );
-
-      //If in available and selected ADD
-      var rolesToAdd = this.availableClientRoles.filter(value =>
+      //If in roles and selected ADD
+      var rolesToAdd = this.clientRoles.filter(value =>
         this.selectedRoles.includes(value)
       );
 
       this.successMessage = "";
       this.errorMessage = "";
 
-      //TODO refactor this
-      UsersRepository.deleteUserClientRoles(
-        this.user.id,
-        this.selectedClientId,
-        rolesToDelete
-      )
-        .then(response => {
-          console.log(response);
-
-          return UsersRepository.addUserClientRoles(
-            this.user.id,
-            this.selectedClientId,
-            rolesToAdd
-          );
-        })
-        .then(response => {
-          console.log(response);
-
-          this.successMessage =
+      Promise.all([
+        UsersRepository.deleteUserClientRoles(
+          this.user.id,
+          this.selectedClientId,
+          rolesToDelete
+        ),
+        UsersRepository.addUserClientRoles(
+          this.user.id,
+          this.selectedClientId,
+          rolesToAdd
+        )
+      ]).then(() => {
+        this.getUserClientRoles();
+        this.successState = true;
+        this.errorState = false;
+        this.successMessage =
             this.successMessage + "Roles Updated Successfully ";
-
-          this.getUserClientRoles();
-        })
-        .catch(error => {
+        window.scrollTo(0, 0);    
+      }).catch(error => {
           this.errorMessage = this.errorMessage + "Error Updating Roles";
```

Comment:
Nitpick or personal preference. I don't know if we have a style guide for this or not, but I feel like the tendency is for modern apps to not use Title Case for messages and alerts.

Gmail:
![image](https://user-images.githubusercontent.com/1767127/84714079-7cc09980-af21-11ea-902a-ad7dd2e020e1.png)

Keycloak:
![image](https://user-images.githubusercontent.com/1767127/84714119-98c43b00-af21-11ea-9d1f-287b742291a5.png)